litellm/proxy/auth/user_api_key_auth.py

```diff
@@ -114,7 +114,11 @@ async def user_api_key_auth(
     )
 
     try:
-        route: str = request.url.path
+        route: str = (
+            request.url.path[len(request.base_url.path) - 1 :]
```

Comment:
this is really confusing code to read - can you write a well named helper function to do this and then use the helper